File: taojieTan/assignments/twopointers/lc713/lc713.js
// Problem Statement #

// Given an array with positive numbers and a target number, find all of its contiguous subarrays whose product is less than the target number.

var numSubarrayProductLessThanK = function (nums, k) {
  // TODO: Write your code here
};

console.log(find_subarrays([2, 5, 3, 10], 30));
console.log(find_subarrays([8, 2, 6, 5], 50));

// Solution
// -----

// var numSubarrayProductLessThanK = function(nums, k) {
//   let count = 0
//   let j = 0
//   let i = 0
//   let product = 1

//   while (i < nums.length && j < nums.length) {
//       if (product * nums[i] < k) {
//           product = product * nums[i]
//           count   = count + (i - j + 1)
//           i++
//       } else {
//           if (nums[j]) product = product / nums[j]
//           j++
//       }
//   }

//   return count
// }

// -----

File: guevaraTorres/assignments/twopointers/lc713/lc713.py
# Problem Statement #

# Given an array with positive numbers and a target number, find all of its contiguous subarrays whose product is less than the target number.

def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
  res = 0
  # TODO: Write your code here
  return res

def main():
  print(find_subarrays([2, 5, 3, 10], 30))
  print(find_subarrays([8, 2, 6, 5], 50))


main()












# Solution
# -----
# def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
#         l = 0
#         res = 0
#         product = 1

#         for r in range(len(nums)):
#             product *= nums[r]

#             if product >= k:
#                 while product >= k and l <= r:
#                     product /= nums[l]
#                     l += 1

#             res += r - l + 1

#         return res

# -----

